### ISSUE TITLE
Unknown command line argument when calling terra_init on OS X

### ISSUE THREAD

**Issue by apoms** [zdevito/terra#114]
When calling terra_init from C++ I am receiving the following error,

```
terra: Unknown command line argument '-x86-asm-syntax=intel'.  Try: 'terra -help'
terra: Did you mean '-nvj-count=intel'?
```

I am statically linking with the Terra library. I think the issue is related to this line: 
https://github.com/zdevito/terra/blob/master/src/tcompiler.cpp#L167

It seems like the llvm command line parser is being called here but I can't understand why that would be. Additionally, the error is being displayed because the command line parser has not been setup to accept any arguments. 

Has anyone successfully used terra_init on OS X with the most recent release?

**Comment by zdevito** [comment]
Can you let me know which LLVM version you are building against? I run Terra with LLVM 3.5 and LLVM 3.6 on OSX 10.10

**Comment by abpoms** [comment]
After doing some more thorough analysis, I've isolated the issue to an interaction with the Halide language library. The following code shows both the failure case and the success case: http://pastebin.com/1dTxDT1D

It seems that if the Halide library is included first then some behavior that Terra relies upon in LLVM is modified (Halide also has an LLVM dependency). If the order is switched then Terra works fine though it might be that Halide is somehow affected.

I have tried building with clang+llvm-3.5.2 and the built in OS X clang/LLVM (version 7.0.0) with equivalent results.
